### ISSUE TITLE
nacos and 6.2.0

### ISSUE THREAD

**Issue by Dauber01** [apache/skywalking#3268]
i used the 6.2.0,use nacos as the cluster,then i run three skywalking oap instance,and in nacos it show like this
![image](https://user-images.githubusercontent.com/28582852/63011118-3beb9480-beba-11e9-8372-bcd65c2f54b8.png)
and one instance throw exception like this,i have clear the elastic search before i update 6.1.0 to 6.2.0

**Comment by wu-sheng** [comment]
Which exception? Do you miss it?

**Comment by Dauber01** [comment]
2019-08-14T09:44:28.692206051Z 2019-08-14 17:44:28,692 - org.apache.skywalking.oap.server.receiver.jvm.provider.handler.JVMSourceDispatcher -51103 [grpc-default-executor-0] WARN  [] - Can't found service by service instance id from cache, service instance id is: 55
2019-08-14T09:44:28.945138635Z 2019-08-14 17:44:28,944 - org.apache.skywalking.oap.server.receiver.jvm.provider.handler.JVMSourceDispatcher -51355 [grpc-default-executor-0] WARN  [] - Can't found service by service instance id from cache, service instance id is: 56
the WARNINGlike this

**Comment by wu-sheng** [comment]
This means you do upgrade in OAP backend, but don't reboot all agents.

**Comment by Dauber01** [comment]
and could  i ask why the nacos show there is only one instance like the picture ↑, but infacet i run three skywalking oap in my server

**Comment by wu-sheng** [comment]
I don't know that for sure. But basically, no matter how many nodes, oap cluster is not working.

**Comment by Dauber01** [comment]
sorrry,i don't hnow what you mean ,is meaning that the the nacos is not working with the 6.2.0 as cluster?

**Comment by Dauber01** [comment]
the nacos is support in 6.2.0 or 6.3.0 .I see the tag 6.2.0 aready have the propeties of nacos

**Comment by wu-sheng** [comment]
I am talking about warning bug. Fix that first.
1. Erase es
1. Reboot agent
1. Reboot oap

Make sure both of all be set.

**Comment by kezhenxu94** [comment]
> and could i ask why the nacos show there is only one instance like the picture ↑, but infacet i run three skywalking oap in my server

@Dauber01 Please make sure that all the three OAP nodes started successfully, if you start all of them in a single server, pay special attention to the buffer file path, which may conflict with each other, resulting only one that can start successfully, as for the nacos cluster implementation, I've tested again, it works as expected

```yml

receiver-trace:
  default:
    bufferPath: ${SW_RECEIVER_BUFFER_PATH:../trace-buffer2/}  # Path to trace buffer files, suggest to use absolute path

# ...

service-mesh:
  default:
    bufferPath: ${SW_SERVICE_MESH_BUFFER_PATH:../mesh-buffer2/}  # Path to trace buffer files, suggest to use absolute path

```

![image](https://user-images.githubusercontent.com/15965696/63022949-98f54380-bed6-11e9-936f-70f9273b2284.png)

**Comment by Dauber01** [comment]
I have start a new node in a new server ,and keep a node up in old server. All of them sart success ,and i start them by docker container. The nacos should show two instance ,but now it show only one too.

**Comment by Dauber01** [comment]
did i need to update the ip in:
core:
  default:
    # Mixed: Receive agent data, Level 1 aggregate, Level 2 aggregate
    # Receiver: Receive agent data, Level 1 aggregate
    # Aggregator: Level 2 aggregate
    role: ${SW_CORE_ROLE:Mixed} # Mixed/Receiver/Aggregator
    restHost: ${SW_CORE_REST_HOST:0.0.0.0}
    restPort: ${SW_CORE_REST_PORT:12800}
    restContextPath: ${SW_CORE_REST_CONTEXT_PATH:/}
    gRPCHost: ${SW_CORE_GRPC_HOST:0.0.0.0}
    gRPCPort: ${SW_CORE_GRPC_PORT:11800}
this config?

**Comment by Dauber01** [comment]
and when i click the detail button ,it show like this:
![image](https://user-images.githubusercontent.com/28582852/63067474-bc090d00-bf41-11e9-8fbc-e94936d9d48d.png)
the ip is 0.0.0.0

**Comment by kezhenxu94** [comment]
**Please make sure all of you OAP nodes start successfully, otherwise they won't register to the Nacos**

## TL;DR

> did i need to update the ip in:
> ```yml
> core:
> default:
> # Mixed: Receive agent data, Level 1 aggregate, Level 2 aggregate
> # Receiver: Receive agent data, Level 1 aggregate
> # Aggregator: Level 2 aggregate
> role: ${SW_CORE_ROLE:Mixed} # Mixed/Receiver/Aggregator
> restHost: ${SW_CORE_REST_HOST:0.0.0.0}
> restPort: ${SW_CORE_REST_PORT:12800}
> restContextPath: ${SW_CORE_REST_CONTEXT_PATH:/}
> gRPCHost: ${SW_CORE_GRPC_HOST:0.0.0.0}
> gRPCPort: ${SW_CORE_GRPC_PORT:11800}
> this config?

If you start two OAP nodes in different machine, then you don't need to modify the IP's and ports here, but if you are starting them in docker containers on the same machine and expose the ports to the host machine, it's highly possible that the ports conflicts, **PLEASE CHECK THE OAP LOG AND MAKE SURE ALL OF THEM START SUCCESSFULLY**

**Comment by Dauber01** [comment]
![image](https://user-images.githubusercontent.com/28582852/63068189-ce387a80-bf44-11e9-9c29-fd28f280e3b1.png)
of course i'm sure the server is srat success:
[Uploading logs.txt…]()
the log is as this

**Comment by Dauber01** [comment]
in nacos every other service show there own ip ,only the oap show the ip of 0.0.0.0

**Comment by kezhenxu94** [comment]
The log file did not upload successfully, @Dauber01 and please upload all the OAP logs, two log files as you said

**Comment by Dauber01** [comment]
[node3.txt](https://github.com/apache/skywalking/files/3504223/node3.txt)
[node2.txt](https://github.com/apache/skywalking/files/3504224/node2.txt)

**Comment by Dauber01** [comment]
like this

**Comment by Dauber01** [comment]
and i find anther thing ,nomater which one insance i start first, the nacos count of instance whill be 1,and when i start another instance in another machine ,the count of instance will not be changed to 2

**Comment by Dauber01** [comment]
![image](https://user-images.githubusercontent.com/28582852/63071762-28403c80-bf53-11e9-8053-c53c72b7aedc.png)

**Comment by kezhenxu94** [comment]
Thanks, I'll dig more into this soon

**Comment by wu-sheng** [comment]
If you are using 0.0.0.0 for all nodes, how do you make service discovery successfully? @kezhenxu94 does nacos cluster coordinator support to set address manually? If not, docker use case may be an issue. For container, k8s coordinator is always better.

**Comment by kezhenxu94** [comment]
@wu-sheng Does it mean that all of the cluster plugins share the same issue when using `0.0.0.0`? Such as zookeeper plugin, it also gets an address `0.0.0.0` in registry

![image](https://user-images.githubusercontent.com/15965696/63074916-2f227b80-bf62-11e9-9b80-1177144fe605.png)

**Comment by wu-sheng** [comment]
When cluster running, 0.0.0.0 is not thr option today. I think that fits all service discovery. Unless coordinator supports auto IP select from VM or manually setting.

**Comment by Dauber01** [comment]
Maybe , i have never read the nacos source code,and if iconfig the ip in core config ,when i use docker ,it can't start up.

**Comment by wu-sheng** [comment]
@Dauber01 Yes. That is not your fault. That is what I am asking @kezhenxu94 to notice. Because as service discovery, you have to know your own IP, but when you run inside docker, it is hard to tell. So whether (a) coordinator provides some locate IP:port config, (b), you have to let the docker knows the IP:port inside.

@Dauber01 What is your container management tech stack?

**Comment by Dauber01** [comment]
k8s coordinator create by rancher

**Comment by wu-sheng** [comment]
@Dauber01 Then use SkyWalking's k8s cluster manager. It is there for a long time, then you don't need to bother all these IP things :)

**Comment by Dauber01** [comment]
thank you ,i will try it

**Comment by 359587** [comment]
How to solve this problem, have you solved it？ @Dauber01
